File: trivially/src/main/java/trivially/ep/m4/finalized/Sub.java
package trivially.ep.m4.finalized;

import trivially.ep.m4.Exp;

public class Sub implements trivially.ep.m4.Sub {

    protected Exp left;

    public Exp getLeft() {
        return this.left;
    }

    protected Exp right;

    public Exp getRight() {
        return this.right;
    }

    public void setLeft(Exp left) { this.left = left; }
    public void setRight(Exp right) { this.right = right; }

    public Sub(Exp left, Exp right) {
        this.left = left;
        this.right = right;
    }
}

File: coco_csharp/CoCo/EP/M2/Finalized/Sub.cs
namespace EP.M2.Finalized {
    public class Sub : Exp, EP.M2.Sub<Exp> {
        private EP.Exp<Exp> left;
        private EP.Exp<Exp> right;

        public Sub(EP.Exp<Exp> _left, EP.Exp<Exp> _right) {
            this.left = _left;
            this.right = _right;
        }

        public EP.Exp<Exp> getLeft() { return this.left; }

        public EP.Exp<Exp> getRight() { return this.right; }
    }
}